### ISSUE TITLE
Details of recent transactions are different in 2.8.2 and 2.8.3 than in 2.7.18 and blockchain.info

### ISSUE THREAD

**Issue by JazzTp** [spesmilo/electrum#2493]
Hi,

I've been off for a while, I switched from 2.7.18 to 2.8.2 and immediately later 2.8.3 was published so I tested with it too.

I can not tell if this is also a server related issue, I can tell that there's no issue here with Electrum 2.7.18, the issue appears with 2.8.2 and 2.8.3.

(I'm running Electrum on Tails 2.12, connecting through the Tor network.)

## THE ISSUE

**The details dialogue of two recent transactions**, dates 2017-05-28 and 2017-06-05, **look different in Electrum 2.8.2 and 2.8.3 than in 2.7.18. Only the latter matches what's in blockchain.info.**

I have passed IDs (plural) of the 2017-05-28 transaction to the main developer by mail a few days ago, I have no problem if he shares or I can pass **(*)** all four IDs of the two transactions through PMs to servers managers like @shsmith if they want to dig into the matter, which for sure keeps me using 2.7.18 for now.

**(*)** both today and tomorrow I will be forcely off line during various hours.

By "different" I mean:

### In Electrum 2.8.2 and 2.8.3

- The ID does not exist in blockchain.info
- The status is "signed" instead of the number of confirmations
- The "broadcast" button is not greyed out (I did **NOT** press it)

### In Electrum 2.7.18

- The ID is matched in blockchain.info
- The status shows as usual before (in earlier dates with earlier Electrum versions) the number of confirmations, matched in blockchain.info
- The "broadcast" button is greyed out

**Previous transactions**, in (2017-26-03 and before) and out (2017-04-04 and before), **look the same** in 2.8.2 and 2.8.3 as in 2.7.18, ID and number of confirmations are matched in blockchain.info and the "broadcast" button is greyed out.

**To rule out a possible problem opening the wallet file, I also tested restoring the wallet from master public keys (and from seed which gave the same identical master public keys). Same identical issue, same IDs.**

**Comment by bauerj** [comment]
Can you send me the signed raw transaction or both txids?

**Comment by JazzTp** [comment]
@bauerj Thanks for replying... sure... I just can't find out how to PM you, at the moment <blush>

**Comment by bauerj** [comment]
You can send an email to bauerj@bauerj.eu.

**Comment by JazzTp** [comment]
Hi,

Sure I had just followed the link on your profile page and found the same
address.



*2017-05-28*~2.6 BTC extraction from btc-e

seen from Electrum 2.8.3 (status "signed" instead of num of confirmations,
and Broadcast button not greyed out, ID not in blockchain.info)
*4234908be060d9e36ea4e7461f951ece4ab8ff7a0c802373758ef61492a54063*

seen from Electrum 2.7.18 (status 1385 confirmations 5/vi/2017 14h13 GMT,
OK in blockchain.info)
*08b9f13f7bfff2c0fb930017141e53645e1efa1fd801fd8e619462b3b9b88d48*



*2017-06-05 *
~1 BTC extraction from btc-e

seen from Electrum 2.8.3 (status "signed" instead of num of confirmations,
and Broadcast button not greyed out, ID not in blockchain.info)
*f5b1b1d080cc6b539a421648acf00bd57146d1a8b85252d8074a210dffc05019*

seen from Electrum 2.7.18 (status 75 confirmations 5/vi/2017 14h1 GNT4, OK
in blockchain.info)
*7d18f0f3ae49b75ae8b0c4e2cee91ee548c6a06fbb7c5f74c53a97f4555bfc31*



Previous transactions, in (2017-26-03 and before) and out (2017-04-04 and
before), appear ok with n confirmations and Broadcast button greyed out.


Again thank you for replying.
Nicola Bernardelli

**Comment by JazzTp** [comment]
P.S.:

"Previous transactions, in (2017-26-03 and before) and out (2017-04-04 and
before), appear OK" *meaning* their details are the same in 2.8.2 and 2.8.3
and 2.7.18, with number of confirmations and Broadcast button greyed out,
and ID matched in blockchain.info.

**Comment by bauerj** [comment]
This seems to be the same issue as in #2462 (since it's also a v2 transaction).

**Comment by JazzTp** [comment]
I see, I searched before posting but I couldn't have told by the title of that issue "Possible regression in transaction hashing/txid calculation" that it is the same issue.

(Might that be the cause for the "signed" status instead of number of confirmations, and "broadcast" button not greyed out?)

Again, 2.7.18 details dialogue is just fine with both transactions.

Also, I should have added, **2.8.3 and 2.8.2 just like 2.7.18 show the grey "check" sign at the left of these two transactions in the history window**. So maybe the hash problem is only happening when opening the details dialogue... just an idea.

**Comment by bauerj** [comment]
No worries, nobody expects you to go through all issues to find something similar 😃 

And yes, this is most likely the cause for the wrong information inside the dialog.

**Comment by JazzTp** [comment]
I see, thank you.

A couple of weeks ago I prepared a Tails + Electrum 2.8.2 setup for a friend as the cold storage.
He opted to use Android Electrum on his cell phone as the watching only part.

He generated a wallet on 2.8.2, passed the Master Public Keys to the watching only part through QR code, and checked with a small sum that he can receive **and** spend from that wallet, everything was working fine.

Can he safely use that setup? I mean, the wallet generated with 2.8.2 should not give any problems as his cold storage, right?

(As I mentioned, I restored my wallet from seed in 2.8.2 and 2.8.3 and checked that I obtained identical Master Public Keys as the ones I get with 2.7.18.)
